Only in dyernew/: __init__.py
diff -aru website-fingerprinting/classifiers/BandwidthClassifier.py dyernew/classifiers/BandwidthClassifier.py
--- website-fingerprinting/classifiers/BandwidthClassifier.py	2018-10-23 17:43:42.769698872 +0100
+++ dyernew/classifiers/BandwidthClassifier.py	2018-10-23 17:42:50.860698872 +0100
@@ -4,7 +4,7 @@
 
 import wekaAPI
 from Packet import Packet
-import arffWriter
+from .. import arffWriter
 
 class BandwidthClassifier:
     @staticmethod
diff -aru website-fingerprinting/classifiers/ESORICSClassifier.py dyernew/classifiers/ESORICSClassifier.py
--- website-fingerprinting/classifiers/ESORICSClassifier.py	2018-10-23 17:43:42.769698872 +0100
+++ dyernew/classifiers/ESORICSClassifier.py	2018-10-23 17:42:50.860698872 +0100
@@ -2,7 +2,7 @@
 # Copyright (C) 2012  Kevin P. Dyer (kpdyer.com)
 # See LICENSE for more details.
 
-import config
+from .. import config
 from Packet import Packet
 import Levenshtein
 
diff -aru website-fingerprinting/classifiers/HerrmannClassifier.py dyernew/classifiers/HerrmannClassifier.py
--- website-fingerprinting/classifiers/HerrmannClassifier.py	2018-10-23 17:43:42.769698872 +0100
+++ dyernew/classifiers/HerrmannClassifier.py	2018-10-23 17:42:50.860698872 +0100
@@ -4,7 +4,7 @@
 
 import wekaAPI
 import math
-import arffWriter
+from .. import arffWriter
 
 # TF-N implementation of Multinomial Naive Bayes Classifier
 class HerrmannClassifier:
diff -aru website-fingerprinting/classifiers/JaccardClassifier.py dyernew/classifiers/JaccardClassifier.py
--- website-fingerprinting/classifiers/JaccardClassifier.py	2018-10-23 17:43:42.769698872 +0100
+++ dyernew/classifiers/JaccardClassifier.py	2018-10-23 17:42:50.860698872 +0100
@@ -2,7 +2,7 @@
 # Copyright (C) 2012  Kevin P. Dyer (kpdyer.com)
 # See LICENSE for more details.
 
-import config
+from .. import config
 
 class JaccardClassifier:
     @staticmethod
diff -aru website-fingerprinting/classifiers/LiberatoreClassifier.py dyernew/classifiers/LiberatoreClassifier.py
--- website-fingerprinting/classifiers/LiberatoreClassifier.py	2018-10-23 17:43:42.769698872 +0100
+++ dyernew/classifiers/LiberatoreClassifier.py	2018-10-23 17:42:50.871698872 +0100
@@ -2,7 +2,7 @@
 # Copyright (C) 2012  Kevin P. Dyer (kpdyer.com)
 # See LICENSE for more details.
 
-import arffWriter
+from .. import arffWriter
 import wekaAPI
 
 class LiberatoreClassifier:
diff -aru website-fingerprinting/classifiers/PanchenkoClassifier.py dyernew/classifiers/PanchenkoClassifier.py
--- website-fingerprinting/classifiers/PanchenkoClassifier.py	2018-10-23 17:43:42.769698872 +0100
+++ dyernew/classifiers/PanchenkoClassifier.py	2018-10-23 17:42:50.871698872 +0100
@@ -3,12 +3,12 @@
 # See LICENSE for more details.
 
 import wekaAPI
-import arffWriter
+from .. import arffWriter
 
-from statlib import stats
+#from statlib import stats
 
-from Trace import Trace
-from Packet import Packet
+from ..Trace import Trace
+from ..Packet import Packet
 import math
 
 class PanchenkoClassifier:
diff -aru website-fingerprinting/classifiers/TimeClassifier.py dyernew/classifiers/TimeClassifier.py
--- website-fingerprinting/classifiers/TimeClassifier.py	2018-10-23 17:43:42.769698872 +0100
+++ dyernew/classifiers/TimeClassifier.py	2018-10-23 17:42:50.871698872 +0100
@@ -4,7 +4,7 @@
 
 import wekaAPI
 from Packet import Packet
-import arffWriter
+from .. import arffWriter
 
 class TimeClassifier:
     @staticmethod
diff -aru website-fingerprinting/classifiers/VNGClassifier.py dyernew/classifiers/VNGClassifier.py
--- website-fingerprinting/classifiers/VNGClassifier.py	2018-10-23 17:43:42.769698872 +0100
+++ dyernew/classifiers/VNGClassifier.py	2018-10-23 17:42:50.871698872 +0100
@@ -2,7 +2,7 @@
 # Copyright (C) 2012  Kevin P. Dyer (kpdyer.com)
 # See LICENSE for more details.
 
-import arffWriter
+from .. import arffWriter
 import wekaAPI
 
 from statlib import stats
diff -aru website-fingerprinting/classifiers/VNGPlusPlusClassifier.py dyernew/classifiers/VNGPlusPlusClassifier.py
--- website-fingerprinting/classifiers/VNGPlusPlusClassifier.py	2018-10-23 17:43:42.769698872 +0100
+++ dyernew/classifiers/VNGPlusPlusClassifier.py	2018-10-23 17:42:50.871698872 +0100
@@ -3,12 +3,12 @@
 # See LICENSE for more details.
 
 import wekaAPI
-import arffWriter
+from .. import arffWriter
 
-from statlib import stats
+#from statlib import stats
 
-from Trace import Trace
-from Packet import Packet
+from ..Trace import Trace
+from ..Packet import Packet
 import math
 
 class VNGPlusPlusClassifier:
diff -aru website-fingerprinting/classifiers/WrightClassifier.py dyernew/classifiers/WrightClassifier.py
--- website-fingerprinting/classifiers/WrightClassifier.py	2018-10-23 17:43:42.769698872 +0100
+++ dyernew/classifiers/WrightClassifier.py	2018-10-23 17:42:50.871698872 +0100
@@ -3,7 +3,7 @@
 # See LICENSE for more details.
 
 import wekaAPI
-import arffWriter
+from .. import arffWriter
 
 class WrightClassifier:
     @staticmethod
Only in dyernew/classifiers: __init__.py
diff -aru website-fingerprinting/classifiers/wekaAPI.py dyernew/classifiers/wekaAPI.py
--- website-fingerprinting/classifiers/wekaAPI.py	2018-10-23 17:43:42.769698872 +0100
+++ dyernew/classifiers/wekaAPI.py	2018-10-23 17:42:50.871698872 +0100
@@ -4,8 +4,8 @@
 
 import re
 import subprocess
-import config
 import os
+from .. import config
 
 def execute( trainingFile, testingFile, classifier, args ):
     myArgs = ["java",
diff -aru website-fingerprinting/config.py dyernew/config.py
--- website-fingerprinting/config.py	2018-10-23 17:43:42.769698872 +0100
+++ dyernew/config.py	2018-10-23 17:44:55.270698872 +0100
@@ -10,17 +10,20 @@
 # * weka-X-Y-Z (see WEKA_ROOT to change the weka version)
 # * pcap-logs (a diretory that contains all of the LL pcap files)
 # * [optional] (a directory that contains custom/local python modules)
-BASE_DIR        = ''
+BASE_DIR        = os.path.expanduser('~')
 
 # Enviromental settings
 JVM_MEMORY_SIZE = '4192m'
 
-WEKA_ROOT          = os.path.join(BASE_DIR   ,'weka-3-7-5')
+WEKA_ROOT          = os.path.join(BASE_DIR   ,'weka-3-8-3/')
 WEKA_JAR           = os.path.join(WEKA_ROOT  ,'weka.jar')
 PCAP_ROOT          = os.path.join(BASE_DIR   ,'pcap-logs')
 PYTHON_ROOT        = os.path.join(BASE_DIR   ,'python2.4')
 PYTHONPATH         = os.path.join(PYTHON_ROOT,'lib/python')
-CACHE_DIR          = './cache'
+CACHE_DIR          = 'attacks/dyer/cache'
+if not os.path.isdir(CACHE_DIR):
+    print '[*] Creating CACHE_DIR in {}'.format(CACHE_DIR)
+    os.makedirs(CACHE_DIR)
 COUNTERMEASURE_DIR = './countermeasures'
 CLASSIFIERS_DIR    = './classifiers'
 OUTPUT_DIR         = './output'
