--- tamaraw.py	2017-08-14 06:59:42.000000000 +0100
+++ tamarawnew/tamaraw.py	2018-10-23 21:15:05.384698872 +0100
@@ -5,6 +5,9 @@
 #Resultant anonymity is measured in ambiguity sizes.
 #Resultant overhead is in size and time.
 #Maximizing anonymity while minimizing overhead is what we want. 
+import os
+import sys
+import shutil
 import math
 import random
 
@@ -103,15 +106,20 @@
 ##    parameters = [100] #padL
 ##    AnoaPad(list2, lengths, times, parameters)
 
-import sys
-import os
+if len(sys.argv) < 2:
+    print 'Usage: {} <input folder>'.format(sys.argv[0])
+    sys.exit(1)
+
+DATASET = sys.argv[1] + '/'
+DEFENDED = 'defended/'
+
 for x in sys.argv[2:]:
     parameters.append(float(x))
 
 sitenum = 100
 instnum = 90
-fold = "batch/"
-foldout = "batch-tamaraw/"
+fold = DATASET
+foldout = DEFENDED
 
 if not os.path.exists(foldout):
     os.makedirs(foldout)
@@ -121,10 +129,10 @@
 anoad = []
 anoadpad = []
 for site in range(0, sitenum):
-    print site
+    print site,
     for inst in range(0, instnum):
         packets = []
-        with open("batch/" + str(site) + "-" + str(inst), "r") as f:
+        with open(fold + str(site) + "-" + str(inst), "r") as f:
             lines = f.readlines()
             starttime = float(lines[0].split("\t")[0])
             for x in lines:
@@ -139,10 +147,9 @@
 
         list3 = []
         
-        AnoaPad(list2, list3, 100, 0)
+        AnoaPad(list2, list3, 750, 0)
 
         fout = open(foldout + str(site) + "-" + str(inst), "w")
         for x in list3:
             fout.write(str(x[0]) + "\t" + str(x[1]) + "\n")
         fout.close()
-
